src/Compression/LZ4_decompress_faster.cpp

```diff
@@ -578,7 +640,13 @@ bool decompress(
     /// Don't run timer if the block is too small.
     if (dest_size >= 32768)
     {
-        size_t best_variant = statistics.select();
+        size_t variant_size = 4;
+#if defined(__AVX512VBMI__) && !defined(MEMORY_SANITIZER)
+        /// best_variant == 4 only valid when AVX512VBMI available
+        if (DB::Cpu::CpuFlagsCache::have_AVX512VBMI)
```

Comment:
Now you have dynamic dispatch but only if the code is compiled with AVX-512 VBMI. But we compile code by default with SSE4.2.

We need to move AVX-512 VBMI code to a separate translation unit, always compile this translation unit with AVX-512 VBMI, and remove `#if defined`.

Reply:
Done, dynamic dispatch with `TargetSpecific`